core/util/google.py

```diff
@@ -77,25 +93,91 @@ def run_crawl(self):
 					req = self.framework.request(url=redirect, allow_redirects=False)
 
 				self._pages += req.text
+				if page == 1:
+					self._first_page += req.text
 				page += 1
 				payload['start'] = set_page(page)
 				if page >= self.limit:
 					break
 
+	@property
+	def google_card_original(self):
+		card_xpath_name = {
+			'card': '//div[@id="wp-tabs-container"]',
+			'card_content': './/div[@class="kno-rdesc"]',
+			'card_info': './/div[@class="rVusze"]'
+		}
+		xpath = {
+			card_xpath_name['card']: [
+				card_xpath_name['card_content'],
+				card_xpath_name['card_info']
+			]
+		}
+		parser = self.framework.page_parse(self._first_page)
+		xpath_results = parser.html_fromstring(xpath)
+		output = {'content': '', 'info': []}
+		root = xpath_results[card_xpath_name['card']]
+		output['content'] = root[card_xpath_name['card_content']][0].text_content()
+		for piece in root[card_xpath_name['card_info']]:
+			output['info'].append(piece.text_content())
+		return output
+
+	@property
+	def results_original(self):
+		parser = self.framework.page_parse(self._pages)
+		xpath_results = parser.html_fromstring(self.xpath_original)
+		results = []
+		if not xpath_results:
+			return results
+		root = xpath_results[self.xpath_name_original['results']]
+		for i in range(len(root[self.xpath_name_original['results_a']])):
+			result = {
+				't': root[self.xpath_name_original['results_title']][i].text_content(),
+				'a': root[self.xpath_name_original['results_a']][i].get('href'),
+				'c': root[self.xpath_name_original['results_cite']][i].text_content(),
+				'd': root[self.xpath_name_original['results_content']][i].text_content(),
+			}
+			results.append(result)
+		return results
+
+	@property
+	def google_card_legacy(self):
```

Comment:
Maybe we need a `results_legacy` property because I am getting this error
![image](https://user-images.githubusercontent.com/33282622/115679420-3dda4c00-a35b-11eb-8b32-955365a68e1e.png)

Reply:
update your google module